File: scripts/thread-pool/thread-pool.py
#!/usr/bin/env python3

import threading
from concurrent.futures import ThreadPoolExecutor

def worker(task_id):
    name = threading.current_thread().name
    print(f"{name} doing task: {task_id}")

if __name__ == '__main__':
    pool = ThreadPoolExecutor(max_workers=5)
    [pool.submit(worker, task_id) for task_id in range(1, 10)]
    pool.shutdown()

File: scripts/thread-pool/thread-pool.pl
#!/usr/bin/env perl

use v5.38;
use threads;
use Thread::Pool;

sub worker($task_id) {
    my $id = threads->self->tid();
    say "Thread $id doing task: $task_id";
}

my $pool = Thread::Pool->new({ workers => 5, do => \&worker });
$pool->job($_) for (1..9);
$pool->shutdown();
